min-turnup/phase1/azure/lib/azure.jsonnet

```diff
@@ -0,0 +1,285 @@
+				}
+			},
+			azurerm_network_security_group: {
+				sg: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: cfg.azure.location,
+					name: names.security_group,
+				},
+			},
+			azurerm_network_security_rule: {
+				[cfg.instance_prefix+"-master-ssh"]: {
+						name: "%(instance_prefix)s-master-ssh" % cfg,
+						priority: 100,
+						direction: "Inbound",
+						access: "Allow",
+						protocol: "Tcp",
+						source_port_range: "*",
+						destination_port_range: "22",
+						source_address_prefix: "*",
+						destination_address_prefix: "*",
+						resource_group_name: "${azurerm_resource_group.rg.name}",
+						network_security_group_name: "${azurerm_network_security_group.sg.name}",
+				},
+				[cfg.instance_prefix+"-master-ssl"]: {
+						name: "%(instance_prefix)s-master-ssl" % cfg,
+						priority: 110,
+						direction: "Inbound",
+						access: "Allow",
+						protocol: "Tcp",
+						source_port_range: "*",
+						destination_port_range: "443",
+						source_address_prefix: "*",
+						destination_address_prefix: "*",
+						resource_group_name: "${azurerm_resource_group.rg.name}",
+						network_security_group_name: "${azurerm_network_security_group.sg.name}",
+				},
+			},
+			azurerm_public_ip: {
+				pip: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: cfg.azure.location,
+					name: names.master_public_ip,
+					public_ip_address_allocation: "static",
+					provisioner:  [{
+						"local-exec": {
+							command: |||
+								cat <<EOF > ../../phase1b/crypto/san-extras
+								DNS.1 = kubernetes
+								DNS.2 = kubernetes.default
+								DNS.3 = kubernetes.default.svc
+								DNS.4 = kubernetes.default.svc.cluster.local
+								DNS.5 = names.master_vm
+								IP.1 = ${azurerm_public_ip.pip.ip_address}
+								IP.2 = 10.3.0.1
+								IP.3 = %s
+								EOF
+							||| % master_private_ip
+						}
+					}]
+				}
+			},
+			azurerm_network_interface: {
+				master_nic: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: "${azurerm_resource_group.rg.location}",
+					name: names.master_nic,
+					ip_configuration: {
+						name: "ipconfig",
+						subnet_id: "${azurerm_subnet.subnet.id}",
+						private_ip_address_allocation: "static",
+						private_ip_address: master_private_ip,
+						public_ip_address_id: "${azurerm_public_ip.pip.id}",
+					},
+					enable_ip_forwarding: true,
+				},
+				node_nic: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: "${azurerm_resource_group.rg.location}",
+					name: names.node_nic+"-${count.index}",
+					ip_configuration: {
+						name: "ipconfig",
+						subnet_id: "${azurerm_subnet.subnet.id}",
+						private_ip_address_allocation: "Dynamic",
+					},
+					enable_ip_forwarding: true,
+					count: cfg.num_nodes
+				}
+			},
+			template_file: {
+				azure_json: {
+					template: "${file(\"azure.json\")}",
```

Comment:
Not sure what you're talking about? This is using terraform's "template_file" functionality. Terraform will interpolate values into the `azure.json` template and then takes the rendered form and uses it as custom_data to the template.

Using jsonnet would require me to somehow write those values out to a file, then execute jsonnet from within terraform, and then regrab that output to pass as custom_data to the VM.

It seems like more work for an unknown reward.